waku/discovery/waku_discv5.nim

```diff
@@ -232,7 +233,18 @@ proc findRandomPeers*(
   elif wd.predicate.isSome():
     discoveredRecords = discoveredRecords.filter(wd.predicate.get())
 
-  waku_discv5_discovered.inc(discoveredRecords.len)
+  # Increment metric for each discovered record's shards
+  for record in discoveredRecords:
```

Comment:
As you said it's computer heavy because we are going through the list of records once returned.

I wonder the metrics could be done in `wd.protocol.queryRandom` call.

I do understand the the `.filter` logic influence that. Which does make me question why we do use `wd.predicate` here, and not within `wd`.